testing/scenario_app/ios/Scenarios/ScenariosTests/AppLifecycleTests.m

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/Flutter.h>
+#import <XCTest/XCTest.h>
+#import "../Scenarios/ScreenBeforeFlutter.h"
+
+@interface AppLifecycleTests : XCTestCase
+@end
+
+@implementation AppLifecycleTests
+
+- (void)setUp {
+  [super setUp];
+  self.continueAfterFailure = NO;
+}
+
+- (void)testLifecycleChannel {
+  XCTestExpectation* engineStartedExpectation = [self expectationWithDescription:@"Engine started"];
+
+  // Let the engine finish booting (at the end of which the channels are properly set-up) before
+  // moving onto the next step of showing the next view controller.
+  ScreenBeforeFlutter* rootVC = [[ScreenBeforeFlutter alloc] initWithEngineRunCompletion:^void() {
+    [engineStartedExpectation fulfill];
+  }];
+
+  [self waitForExpectations:[NSArray arrayWithObject:engineStartedExpectation] timeout:5];
```

Comment:
You don't need to specify the expectation, you can use `-[self waitForExpectationsWithTimeout:5.0:handler:nil]`